### PR TITLE
Authentication pages added, input component refactored

### PR DESCRIPTION
**Pull request by Sylpherena**

### CODE REVIEW

File: src/components/Layout/ThemeSelector.tsx [hunk 1]
```diff
@@ -11,15 +11,21 @@ export default function ThemeSelector() {
     setTheme(selectedTheme);
   };
 
+  const formatThemeName = (name: string) => {
+    return name.charAt(0).toUpperCase() + name.slice(1);
```
Comment (hsynaksu) [inline src/components/Layout/ThemeSelector.tsx:15]: camel case can be shorter

File: src/components/Layout/ThemeSelector.tsx [hunk 2]
```diff
@@ -28,7 +34,7 @@ export default function ThemeSelector() {
               type="radio"
               name="theme-dropdown"
               className="theme-controller btn btn-sm btn-block btn-ghost justify-start"
-              aria-label={t.charAt(0).toUpperCase() + t.slice(1)}
+              aria-label={formatThemeName(t)}
```
Comment (hsynaksu) [inline src/components/Layout/ThemeSelector.tsx:46]: is using aria-label correct?

File: src/components/PasswordInput.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    return (
+      <Input
+        Icon={Icon ?? KeyRound}
+        placeholder={placeholder}
+        type={isPasswordShown ? "text" : "password"}
+        endSlot={
+          <label className="swap swap-rotate z-20">
+            {/* this hidden checkbox controls the state */}
+            <input type="checkbox" onClick={(e) => handleShowPassword(e)} />
```
Comment (hsynaksu) [inline src/components/PasswordInput.tsx]: aria-label would be good to add

Comment (hsynaksu) [reply src/components/PasswordInput.tsx]: belki input yerine swap-active kullanılabilir

File: src/index.css [hunk 4]
```diff
@@ -1,3 +1,17 @@
 @tailwind base;
 @tailwind components;
 @tailwind utilities;
+
```
Comment (hsynaksu) [inline src/index.css:4]: add comment if you can

File: src/pages/Authentication/Authentication.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import { useParams } from "react-router-dom";
+import AuthenticationTabs from "./AuthenticationTabs";
+import ForgotPassword from "./Forms/ForgotPassword";
+
+export default function Authentication() {
+  const { tab } = useParams();
```
Comment (hsynaksu) [inline src/pages/Authentication/Authentication.tsx]: can be seperator to different routes

File: src/pages/Authentication/Forms/ForgotPassword.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        >
+          <ChevronLeft />
+        </button>
+        <div className="flex flex-col gap-4">
+          <h3 className="text-center text-3xl font-bold text-base-content">
+            Forgot Password?
+          </h3>
+          <p className="text-center label">
+            Enter your registered e-mail and password reset link will be sent to
+            that e-mail.
```
Comment (hsynaksu) [inline src/pages/Authentication/Forms/ForgotPassword.tsx:48]: unnecessary new line

File: src/pages/Authentication/Forms/Signin.tsx [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+  SignInFormType,
+  signInSchema,
+} from "./authenticationSchemas";
+import { zodResolver } from "@hookform/resolvers/zod";
+
+export default function Signin() {
+  const {
+    register,
+    handleSubmit,
+
```
Comment (hsynaksu) [inline src/pages/Authentication/Forms/Signin.tsx]: line

File: src/pages/Authentication/Forms/authenticationSchemas.ts [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+export interface ForgotEmailFormType {
+  email: string;
+}
+
+export const defaultForgotEmailFormValues = {
+  email: "",
+};
+
+export interface SignInFormType extends ForgotEmailFormType {
```
Comment (hsynaksu) [inline src/pages/Authentication/Forms/authenticationSchemas.ts]: should not be coupled, different types

File: src/components/Layout/ThemeSelector.tsx (hunk 1 above)
Comment (Sylpherena) [reply src/components/Layout/ThemeSelector.tsx:15]: camel case function is not a default function, I dont have lodash in this project

File: src/components/Layout/ThemeSelector.tsx (hunk 2 above)
Comment (Sylpherena) [reply src/components/Layout/ThemeSelector.tsx:46]: daisy ui adds aria-label as content

File: src/index.css (hunk 4 above)
Comment (Sylpherena) [reply src/index.css:4]: commented

File: src/pages/Authentication/Forms/ForgotPassword.tsx (hunk 6 above)
Comment (Sylpherena) [reply src/pages/Authentication/Forms/ForgotPassword.tsx:48]: prettier auto-fix

File: src/pages/Authentication/Forms/Signin.tsx (hunk 7 above)
Comment (Sylpherena) [reply src/pages/Authentication/Forms/Signin.tsx]: fixed

File: src/pages/Authentication/Authentication.tsx (hunk 5 above)
Comment (Sylpherena) [reply src/pages/Authentication/Authentication.tsx]: added params as different routes

File: src/components/PasswordInput.tsx (hunk 3 above)
Comment (Sylpherena) [reply src/components/PasswordInput.tsx]: used swap-active and added aria-labels to whole project where needed

File: src/pages/Authentication/Forms/authenticationSchemas.ts (hunk 8 above)
Comment (Sylpherena) [reply src/pages/Authentication/Forms/authenticationSchemas.ts]: seperated the types

**Review by hsynaksu — APPROVED** [review]

**Pull request merged**